stencil-push.sh

```diff
@@ -0,0 +1,58 @@
+set -e
+
+project_version=$1
+proto_repo=$2
+file_path=$3
+output_socket=/dev/null
+
+log_info()  { echo "$(date) - [INFO] $@"; }
+log_warn()  { echo "$(date) - [WARN] $@"; }
+log_error() { echo "$(date) - [ERROR] $@"; }
+
+help() {
+    echo ""
+    echo "stencil-push.sh <version> <proto-repo> <file-path>"
+    echo "   version      - version of the proto descriptor set being uploaded"
+    echo "   proto-repo   - repository name where to upload the proto descriptor set"
+    echo "   file-path    - path to the proto descriptor set binary file"
+    echo "   "
+    echo "   stencil-push.sh needs a few ENV variables set - STENCIL_HOSTNAME, STENCIL_USERNAME, STENCIL_PASSWORD"
+    echo "   STENCIL_HOSTNAME - Hostname serving the stencil APIs"
+    echo "   STENCIL_USERNAME - HTTP Basic Auth username"
+    echo "   STENCIL_PASSWORD - HTTP Basic Auth password"
+}
+
+if [[ -z "$STENCIL_HOSTNAME" ]] || [[ -z "$STENCIL_USERNAME" ]] || [[ -z "$STENCIL_PASSWORD" ]]
```

Comment:
I think the user shouldn't be forced to provide a user/pwd. This script should deal this as a provision and therefore if it's provided process them, if not simply curl without. Makes it generic, WDYT?

Reply:
These docs are about a specific implementation as of now. I think a shell script is a basic start for our automation but are expecting a full fledged gradle plugin for future.
That would make it easier to support all kinds of auth mechanisms as well as any other kind of customisations that might arise due to separate setup.